mood-analysis.rb

```diff
@@ -28,5 +29,60 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+def strip_punctuation(words)
+  # 2. To make the results a little more accurate, let's write and utilize a method called `strip_punctuation` to 
+  # strip out the punctuation that affects the results. Namely, remove  exclamation marks (!), periods (.), commas (,), 
+  # and hashtags (#).
+  words.gsub(/[^\/A-Za-z0-9\s]/i, "")
+end
+
+# "smiley collector": iterate a corpus, analyze its mood, store return in an array.
+def print_all_moods(words)
+  words.collect { |statement| statement[0..4] + " " + analyze_mood(statement)}
+end
+
+def happy_days(words)
+  # Write a method called `happy_days` to determine how many logged entries it takes until there have 
+  # been three :-) happy days.
+
+  total_days = 0
+  hap_days = 0
+
+  words.each do |statement|
+    total_days += 1
+    if analyze_mood(strip_punctuation(statement)) == ":-)"
+      hap_days += 1
+    end
+    break if hap_days == 3
```

Comment:
There's no reason to break here and then later return this information upon checking the condition again. Instead you could have just inserted your return right here.